### DISCUSSION TITLE
Bricking both V2 Pan and V3 after editing wz_mini

### DISCUSSION THREAD

**Discussion by greekgod1820** [gtxaspec/wz_mini_hacks#423] [Q&A]
Alight hoping some others have insight here are the steps that cause my issue:

1) Open new V3 cam and downgrade firmware to 4.36.9.139 manually via microSD card
2) Setup camera and stream live feed, confirm firmware is in fact 4.36.9.139
3) Format SD card to FAT32 and copy contents of SD_Root to root of SD card
4) With cam off insert SD card and plug camera in powering it on
5) Confirm cam is still working post boot via Wyze App
6) Reboot cam via app and confirm still working
7) Unplug camera from wall remove SD card make appropriate mods to wz_mini.  I'm only modifying the RSTP stream part with these parameters

#####VIDEO STREAM#####
RTSP_LOGIN="xyz"
RTSP_PASSWORD="123"
RTSP_PORT="8554"

RTSP_HI_RES_ENABLED="true"
RTSP_HI_RES_ENABLE_AUDIO="true"
RTSP_HI_RES_FPS="2048"
RTSP_HI_RES_MAX_BITRATE="1024"
RTSP_HI_RES_TARGET_BITRATE="2"
RTSP_HI_RES_ENC_PARAMETER=""

RTSP_LOW_RES_ENABLED="false"
RTSP_LOW_RES_ENABLE_AUDIO="false"
RTSP_LOW_RES_FPS=""
RTSP_LOW_RES_MAX_BITRATE=""
RTSP_LOW_RES_TARGET_BITRATE=""
RTSP_LOW_RES_ENC_PARAMETER=""

RTSP_AUTH_DISABLE="false"

8) Save the file via notepad ++ and eject SD card
9) With camera unplugged insert SD card back into camera
10) Camera powers but red light turns on remains solid.

No matter what I do at this point it remains solid.  Manual firmware flash, removing SD card, nothing it stay solid no matter what.  This has happened to two V3s and a Pan V2.  What am I doing wrong here?

**Comment by Bob-AOL** [comment]
FPS should be 20 with the remaining entries moved down one position each. I'd also change the RTSP port to 554 just because that's what it normally is in the real world.

**Reply by greekgod1820** [reply]
tried still solid red light with no responses in wyze app, tiny cam, or activity on my wifi for it

**Comment by greekgod1820** [comment]
You think that’s what is causing the cameras to brick?  Even after removing
the SD card rebooting holding setup button anything light stays solid red.

**Comment by Bob-AOL** [comment]
Have you tried re-flashing the original firmware by the normal procedure? Insert SD card unpowered. Re-power while holding reset button for several seconds until light changes to blue/purple....

https://support.wyze.com/hc/en-us/articles/360031490871-How-to-flash-your-Wyze-Cam-firmware-manually

**Comment by greekgod1820** [comment]
Yes did that, tried multiple firmware’s as well and multiple SD cards.  All
formatted FAT-32

**Comment by Bob-AOL** [comment]
Have you made the corrections to the .conf file and tried it? And try using plain old notepad to edit your .conf file.

**Reply by greekgod1820** [reply]
Just tried still solid red light with no response

**Comment by Bob-AOL** [comment]
This may seem like a dumb question, but are you using a properly rated power supply?

**Reply by greekgod1820** [reply]
Using stock power supply also tried one from a fully functioning camera.

**Comment by Bob-AOL** [comment]
And an even dumber question..., is the camera status light perhaps enabled in the advanced settings and the cameras are actually working?

**Reply by greekgod1820** [reply]
Camera is not responsive shows as offline in wyze app, does not appear in tinycam as online, and shows no connection to my wifi router in the device list.

**Reply by greekgod1820** [reply]
Also no questions are dumb when providing support trust me I've hear far worse when helping others :)

**Comment by Bob-AOL** [comment]
Backtrack to when it worked and re-trace your steps again.

SD card aside, keep trying to re-initialize by the reset button approach a few more times without the SD card. You may get lucky there. I'm out of bullets on this one. I don't see how your approach as explained would brick three cameras -- or why you didn't stop to figure it out after the first one!

Key to re-flashing is having the reset button pushed in _before_ applying power and keeping it held in until you get the purple/blue light. Release and let it do its thing until it reboots. Remove SD card and power back up.

Don't know what else to suggest.

**Reply by greekgod1820** [reply]
No worries I assumed first camera was simply defective.  Second camera worked fine so I did the third.  Then went back in to update the config files to not be the stock one and BOM both cameras did as the first did.  So with a brand new camera I'm able to downgrade the firmware, copy root contents to the SD card get the audio for swap going and get the cameras up and running but the second I power down take the card out to modify the wz_mini file I lose the camera to the solid red light issue.  Holding the setup prior to powering with SD card in for manual firmware flash doesn't even work which is concerning.  These are all within return period so now I'm on that same bucket, figure out what is bricking them or return in a week or so before return period ends.

**Comment by Bob-AOL** [comment]
Is this a DHCP situation without any address reservations, or could there possibly be an IP conflict hosing things up?

**Reply by greekgod1820** [reply]
Very good thought as well but I have IP reservations setup and assigned to the MAC address.  But I figured reboot anyway just to be sure still same.

**Comment by Bob-AOL** [comment]
Kill the IP reservations for starters. And try a bare bones wz_mini conf. file with no bells and whistles -- especially any ethernet stuff enabled. That kills your wifi and any dhcp discovery activity until the ethernet adapter is recognized and configured  by a separate process -- if you're going that route.

Pretty sure this is something simple.

**Reply by greekgod1820** [reply]
Sorry should have mentioned I tried that.  I'm able with a brand new camera to get it up and running with a stock wz_mini conf file with no mods at all.  What really concerns me is that the camera is unresponsive even after powering down and removing the SD card.  Something is happening or getting stuck in temp. memory that locks it up preventing it from just powering up via normal wyze firmware, preventing manual flashing of the firmware etc.  I get the purple light when i try to manually flash but it goes solid red after a while.  I've left it as long as 2hrs in that state and it's never progressed.

**Comment by Bob-AOL** [comment]
Yes, it sounds like it's hung in between stock and hacked and needs a kick start. There may be logs with clues on the SD card(s) of the failed cams.

In the meantime...

Doesn't get much more basic than this. _Killing the audio may be key,_ Enabling the web server allows editing and saving of the .conf file on the SD card while in the camera.  Just don't reboot from the web interface. Use the app to reboot.

This is an earlier bare bones .conf just to strip everything down to minimum essentials. Give it a spin and see what happens.

#wz_mini configuration


#### W E B CAMERA###
##THIS MODE DISABLES EVERYTHING AND IT WILL
## WORK AS A WEB CAMERA FOR YOUR PC ***ONLY***
WEB_CAM_ENABLE="false"
WEB_CAM_BIT_RATE="8000"
WEB_CAM_FPS_RATE="25"

#####NETWORKING#####
CUSTOM_HOSTNAME="V2_Mini"

ENABLE_USB_ETH="false"
ENABLE_USB_ETH_MODULE_AUTODETECT="true"
ENABLE_USB_ETH_MODULE_MANUAL=""

ENABLE_USB_DIRECT="false"
USB_DIRECT_MAC_ADDR="02:01:02:03:04:08"

ENABLE_USB_RNDIS="false"

ENABLE_IPV6="false"

ENABLE_WIREGUARD="false"
WIREGUARD_IPV4=""
WIREGUARD_PEER_ENDPOINT=""
WIREGUARD_PEER_PUBLIC_KEY=""
WIREGUARD_PEER_ALLOWED_IPS=""
WIREGUARD_PEER_KEEP_ALIVE=""

ENABLE_IPTABLES="false"

ENABLE_NFSv4="false"

ENABLE_RTL8189FS_DRIVER="true"
ENABLE_ATBM603X_DRIVER="true"

##### NETWORK INTERFACE BONDING #####
BONDING_ENABLED="false"
BONDING_PRIMARY_INTERFACE="eth0"
BONDING_SECONDARY_INTERFACE="wlan0"
BONDING_LINK_MONITORING_FREQ_MS="100"
BONDING_DOWN_DELAY_MS="5000"
BONDING_UP_DELAY_MS="5000"
BONDING_FAIL_OVER_MAC="0"

#####ACCESSORIES#####
REMOTE_SPOTLIGHT="false"
REMOTE_SPOTLIGHT_HOST="0.0.0.0"

#####VIDEO STREAM#####
RTSP_LOGIN="user"
RTSP_PASSWORD="userpass"
RTSP_PORT="554"

RTSP_HI_RES_ENABLED="true"
RTSP_HI_RES_ENABLE_AUDIO="false"
RTSP_HI_RES_FPS="15"
RTSP_HI_RES_MAX_BITRATE="2048"
RTSP_HI_RES_TARGET_BITRATE="1024"
RTSP_HI_RES_ENC_PARAMETER="2"

RTSP_LOW_RES_ENABLED="false"
RTSP_LOW_RES_ENABLE_AUDIO="false"
RTSP_LOW_RES_FPS=""
RTSP_LOW_RES_MAX_BITRATE=""
RTSP_LOW_RES_TARGET_BITRATE=""
RTSP_LOW_RES_ENC_PARAMETER=""

RTSP_AUTH_DISABLE="false"

#####RTMP STREAMING#####
#### REQUIRES RTSP ####
### AND RTSP AUDIO ###
RTMP_STREAM_ENABLED="false"
RTMP_STREAM_FEED="video1_unicast"
RTMP_STREAM_SERVICE="youtube"
RTMP_STREAM_DISABLE_AUDIO="false"
RTMP_STREAM_YOUTUBE_KEY=""
RTMP_STREAM_TWITCH_KEY=""
RTMP_STREAM_FACEBOOK_KEY=""

#####GENERAL#####
LIBCALLBACK_ENABLE="true"
ENABLE_SWAP="true"
ENABLE_USB_STORAGE="false"
ENABLE_EXT4="false"
ENABLE_CIFS="false"
DISABLE_FW_UPGRADE="false"
AUDIO_PROMPT_VOLUME="50"
ENABLE_MP4_WRITE="false"
NIGHT_DROP_DISABLE="false"
DISABLE_MOTOR="false"
ENABLE_FSCK_ON_BOOT="false"
ENABLE_CAR_DRIVER="false"
ENABLE_LOCAL_DNS="false"
ENABLE_CRONTAB="false"

#####DEBUG#####
#drops you to a shell via serial, doesn't load app_init.sh
DEBUG_ENABLED="false"
DEBUG_INITRAMFS_ENABLED="false"
DEBUG_PASSWORD="false"
ENABLE_SYSLOG_SAVE="false"

####WEB####
WEB_SERVER_ENABLED="true"

#####SCRIPTING#####
CUSTOM_SCRIPT_PATH=""

**Comment by Bob-AOL** [comment]
I need to depart for now. Good luck.

**Comment by jmbulan** [comment]
i ran into this situation after updating the "camera settings" thru the httpd. the fix had something to do with modifying the s14 file and deleting onetimerun. then reflashing the card...  ill try looking it up when i get home

**Reply by greekgod1820** [reply]
Any help would be appreciated I'd like to figure this one out encase others run into the issue as well.

**Comment by gtxaspec** [comment]
@greekgod1820 when you try to flash the stock firmware, do you get the purple lights after holding down setup with the firmware on the sd card?

**Comment by greekgod1820** [comment]
I do, for what to my experience is the right amount of time.  But then
instead of flashing red when it’s all done it goes solid red.

**Reply by Bob-AOL** [reply]
As you have come to realize, some of these issues move along rather slowly and require a certain degree of patience before receiving much assistance, let alone solutions. I'm playing around with a couple new V3s and the latest iteration of wz_mini myself and am finding it problematic in a couple of areas compared to the version I cut my teeth on last summer.

I haven't bricked anything yet, but am having trouble in several areas -- not the least of which is a time/date issue with the .conf timestamp that is 5 hours ahead of current time when using the web interface for editing and updating the configuration. That couldn't be creating other problems, could it? 

More later..., probably in a new thread, but I fear the project has gotten a bit too fat and lost some of the fundamental focus that was at its heart in the beginning. I'm certainly not criticizing what has been developed as it's some great work. It would just ne nice if some of the more fundamentally intended functionality received as much support as the newer peripheral ideas that may only appeal to the interests of a select few.

Most folks just want a USB ethernet solution that's easy to configure and works. 

BTW ,try letting off the reset button as soon as it goes purple. Keeping it depressed could be causing some confusion to the process. Just thinking out loud...

**Reply by greekgod1820** [reply]
@Bob-AOL I tried your suggestion as I could see that being something silly enough to solve the issue.  It unfortunately did not work to solve it as well.  I've got no issue with the pace of resolving the issue.  I unfortunately did have to return the unit for exchange as the return period was ending.  However, I can easily replicate the issue clearly for continued testing ;).

**Comment by virmaior** [comment]
@Bob-AOL  ` time/date issue with the .conf timestamp that is 5 hours ahead of current time when using the web interface ` could be UTC.

@greekgod1820  `Unplug camera from wall remove SD card make appropriate mods to wz_mini.` what application are you using to edit this?

**Reply by Bob-AOL** [reply]
Could the time stamp cause a problem and how would I correct it?

**Reply by greekgod1820** [reply]
@virmaior I did not use the web interface to make edits to the file I always powered down the camera and put the micro SD back into my computer.  If that would be the case as well in my scenario I could replicate my problem again and give it a try.  I used notepad on windows 10 to make the edits.  I then on a second camera used notepad ++ both produced the same results.

**Comment by Bob-AOL** [comment]
I've found the web interface to be quite convenient and efficient in editing the config and rebooting while staying online -- which is a requisite of successfully enabling the USB/Ehernet feature -- until my last attempts of yesterday and today. It simply kills the wifi and wz_mini seems to refuse to load thereafter. I've become too frustrated to dig very deep into whatever logs may offer clues to what has changed since the last time did it.

**Reply by Rustymage** [reply]
It seems I may be suffering the same issues- https://github.com/gtxaspec/wz_mini_hacks/issues/442

**Reply by greekgod1820** [reply]
Read through your thread does sound like the same failure symptoms.  Only difference really my edits were performed via notepad and notepad++ so local not via the web gui so the GUI reboot was not utilized.  I did make similar edits otherwise.

**Reply by Bob-AOL** [reply]
If you don't enable USB/ETH as a live edit and reboot (not power cycle) with the camera online, it won't work and just kills the camera. 

Using an external editor to activate the USB/ETH feature disables the wifi driver necessary for the camera to boot in the first place. It doesn't know there's an ethernet adapter even there yet unless it can actively discover and initialize it first. THEN it kills the wifi and heads down the road on its merry way as an ethernet device. The Readme explains it, but not as clearly as should be emphasized for folks who seem to think just hanging an adapter off the back of a wz_mini v3 is plug'n'play -- which I have actually seen claimed and is misleading.

**Reply by greekgod1820** [reply]
@Bob-AOL I actually don’t enable the USB/ETH at all as I don’t use it.  I only enable RTSP, Disable firmware update.  I ended up exchanging the solid red light ones before my time was up.  Based on the suggestions of using the web GUI with the replacements I followed the same steps as I outlined except enabled web gui and made all my edits via the GUI first round.  All three 2 pan v2 and 1 v3 are all running fine streaming rtsp feeds for me.

**Reply by Bob-AOL** [reply]
Copy.

**Reply by Bob-AOL** [reply]
I just finished re-flashing my beat to hell bench v3 with RTSP 4.61.0.3 and running wz_mini to set up another UCTRONICS adapter. I used the web GUI to enable the USB/ETH function and for subsequent reboot. All the wz_mini.conf settings otherwise were done previously in notepad on my Win10 laptop. 

It worked perfectly and the backup was properly maintained -- this time.
